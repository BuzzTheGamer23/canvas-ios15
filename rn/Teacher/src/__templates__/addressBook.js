/* @flow */

import template, { type Template } from '../utils/template'

export const addressBookResult: Template<AddressBookResult> = template({
  id: 1,
  name: 'Donald Trump',
  common_courses: {
    '11': ['TeacherEnrollment'],
  },
})
