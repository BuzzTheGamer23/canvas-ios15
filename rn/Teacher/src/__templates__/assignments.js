//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const assignment: Template<Assignment> = template({
  id: 123456789,
  course_id: 987654321,
  name: 'learn to write code',
  description: 'a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
  created_at: '2017-03-17T19:15:25Z',
  updated_at: '2017-03-17T19:15:25Z',
  points_possible: 20,
  grading_type: 'points',
  assignment_group_id: 1376,
  peer_reviews: false,
  automatic_peer_reviews: false,
  position: 1,
  submission_types:
  [ 'online_text_entry',
    'online_url',
    'media_recording',
    'online_upload' ],
  needs_grading_count: 0,
  html_url: 'http://hostname.tld/whatever',
  overrides: [],
})

export const assignmentGroup: Template<AssignmentGroup> = template({
  id: 1,
  name: 'Learn React Native',
  position: 1,
  assignments: [assignment()],
})

export const assignmentDueDate: Template<AssignmentDate> = template({
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})

export const assignmentOverride: Template<assignmentOverride> = template({
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})
