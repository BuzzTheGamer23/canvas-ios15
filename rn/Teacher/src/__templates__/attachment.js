/* @flow */

import template, { type Template } from '../utils/template'
import { file } from './file'

export const attachment: Template<Attachment> = template({
  ...file(),
  filename: 'Attachment 1.jpg',
  uri: 'file://somewhere/on/disk.jpg',
})
