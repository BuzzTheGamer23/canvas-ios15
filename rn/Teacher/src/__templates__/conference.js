//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const conference: Template<Conference> = template({
  id: '1',
  title: 'super sabido sensacional',
  join_url: 'https://wwww.instructure.com/courses/1/conference/1',
  description: 'this is going to be a great conference',
})
