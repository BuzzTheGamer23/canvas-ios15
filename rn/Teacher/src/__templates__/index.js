// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './folder'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './page'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './toDo'
export * from './users'
