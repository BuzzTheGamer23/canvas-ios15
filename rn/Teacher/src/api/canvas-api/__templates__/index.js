/* @flow */

export default {
  ...require('./course'),
}
