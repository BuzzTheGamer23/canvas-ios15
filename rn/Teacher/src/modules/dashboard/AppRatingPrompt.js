//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import i18n from 'format-message'
import React from 'react'
import {
  Image,
  LayoutAnimation,
  StyleSheet,
  TouchableHighlight,
  View,
  NativeModules,
} from 'react-native'
import images from '../../images'
import colors from '../../common/colors'
import { Heading2 } from '../../common/text'
import { Button } from '../../common/buttons'
import DashboardContent from './DashboardContent'
import { logEvent } from '../../common/CanvasAnalytics'

type Props = NavigationProps & {
  collapsed: boolean,
  style: ?{[string]: any},
}

type State = {
  collapsed: boolean,
}

export default class AppRatingPrompt extends React.Component<Props, State> {
  state = {
    collapsed: this.props.collapsed,
  }

  componentWillReceiveProps (newProps: Props) {
    this.state.collapsed = newProps.collapsed
  }

    dismiss = () => {
      NativeModules.AppStoreReview.handleUserFeedbackOnDashboard(false)
      this.hide()
    }

    hide = () => {
      LayoutAnimation.easeInEaseOut()
      this.setState({ collapsed: true })
    }

    render () {
      const { style } = this.props
      let { collapsed } = this.state

      return (
        <DashboardContent
          style={[style, (collapsed ? styles.collapsed : styles.expanded)]}
          contentStyle={[{ borderColor: 'transparent', borderWidth: 0 }]}
          hideShadow={true}
        >
          <View style={styles.rowContent}>
            <TouchableHighlight
              accessibilityTraits='button'
              accessibilityLabel={i18n(`Dismiss prompt to rate`)}
              onPress={this.dismiss}
              underlayColor='transparent'
              style={styles.closeButtonContainer}
              testID={`prompt-to-rate.butotn.dismiss`}
            >
              <Image source={images.x} style={styles.dismissIcon} />
            </TouchableHighlight>
            <Heading2 style={styles.header}> {i18n('Are you enjoying Canvas?')} </Heading2>
          </View>

          <View style={styles.buttonContainer}>
            <Button
              onPress={() => this.handleResponse(false)}
              style={[styles.buttonText]}
              containerStyle={[styles.button]}
              testID={`prompt-to-rate.button.no`}
              accessibilityTraits='button'
            >
              {i18n('No')}
            </Button>
            <Button
              onPress={() => this.handleResponse(true)}
              style={styles.buttonText}
              containerStyle={styles.button}
              testID={`prompt-to-rate.button.yes`}
              accessibilityTraits='button'
            >
              {i18n('Yes')}
            </Button>
          </View>
        </DashboardContent>
      )
    }

    handleResponse = (userAcceptedAppReview: boolean) => {
      let event
      NativeModules.AppStoreReview.handleUserFeedbackOnDashboard(userAcceptedAppReview)
      if (userAcceptedAppReview) {
        event = 'appReview_userAccepted'
      } else {
        event = 'appReview_userDeclined'
        this.promptUserForFeedback()
      }
      logEvent(event)
      this.hide()
    }

    promptUserForFeedback = () => {
      this.props.navigator.show('/support/problem', { modal: true })
    }
}

const styles = StyleSheet.create({
  closeButtonContainer: {
    position: 'absolute',
    right: 0,
    top: 0,
    padding: 12,
  },
  header: {
    textAlign: 'center',
    fontWeight: '600',
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    flexWrap: 'wrap',
    margin: 8,
  },
  button: {
    flex: 1,
    height: 40,
    borderRadius: 4,
    margin: 4,
    backgroundColor: 'white',
    borderColor: colors.grey4,
    borderWidth: StyleSheet.hairlineWidth,
  },
  buttonText: {
    color: colors.link,
    fontSize: 16,
    fontWeight: '600',
  },
  rowContent: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    padding: 8,
  },

  dismiss: {
    alignSelf: 'flex-end',
    paddingHorizontal: 4,
    paddingVertical: 2,
  },
  dismissIcon: {
    width: 16,
    tintColor: colors.grey4,
  },
  collapsed: {
    height: 0,
    overflow: 'hidden',
  },
  expanded: {
    overflow: 'hidden',
  },
})
