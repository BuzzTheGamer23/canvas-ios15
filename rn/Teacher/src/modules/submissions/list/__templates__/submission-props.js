// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
})
