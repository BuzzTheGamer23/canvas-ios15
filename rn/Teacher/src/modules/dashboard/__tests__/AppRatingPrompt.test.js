//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

// @flow

import { shallow } from 'enzyme'
import React from 'react'
import { NativeModules } from 'react-native'
import AppRatingPrompt from '../AppRatingPrompt'
import * as templates from '../../../__templates__'

describe('AppRatingPrompt', () => {
  const defaults = {
    style: { width: 320, padding: 8 },
    collapsed: false,
    navigator: templates.navigator(),
  }

  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('renders', () => {
    const tree = shallow(
      <AppRatingPrompt {...defaults} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders hidden', () => {
    const tree = shallow(
      <AppRatingPrompt {...defaults} collapsed={true} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('can like app', () => {
    const tree = shallow(
      <AppRatingPrompt {...defaults} />
    )
    tree.find('[testID="prompt-to-rate.button.yes"]').simulate('Press')
    expect(NativeModules.AppStoreReview.handleUserFeedbackOnDashboard).toHaveBeenCalledWith(true)
    expect(NativeModules.CanvasAnalytics.logEvent).toHaveBeenCalledWith('appReview_userAccepted', undefined)
  })

  it('can not like app', () => {
    const tree = shallow(
      <AppRatingPrompt {...defaults} />
    )
    tree.find('[testID="prompt-to-rate.button.no"]').simulate('Press')
    expect(NativeModules.AppStoreReview.handleUserFeedbackOnDashboard).toHaveBeenCalledWith(false)
    expect(NativeModules.CanvasAnalytics.logEvent).toHaveBeenCalledWith('appReview_userDeclined', undefined)
    expect(defaults.navigator.show).toHaveBeenCalledWith('/support/problem', { 'modal': true })
  })

  it('can hide', () => {
    const props = {
      ...defaults,
    }
    const tree = shallow(
      <AppRatingPrompt {...props} />
    )
    tree.find('[testID="prompt-to-rate.butotn.dismiss"]').simulate('Press')
    expect(tree).toMatchSnapshot()
  })

  it('dismiss pressed', () => {
    const tree = shallow(
      <AppRatingPrompt {...defaults} />
    )
    tree.find('[testID="prompt-to-rate.butotn.dismiss"]').simulate('Press')
    expect(NativeModules.AppStoreReview.handleUserFeedbackOnDashboard).toHaveBeenCalledWith(false)
    expect(tree.instance().state.collapsed).toBe(true)
  })
})
