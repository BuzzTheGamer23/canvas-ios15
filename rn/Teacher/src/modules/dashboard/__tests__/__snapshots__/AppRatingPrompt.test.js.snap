// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRatingPrompt can hide 1`] = `
<Component
  contentStyle={
    Array [
      Object {
        "borderColor": "transparent",
        "borderWidth": 0,
      },
    ]
  }
  hideShadow={true}
  style={
    Array [
      Object {
        "padding": 8,
        "width": 320,
      },
      Object {
        "height": 0,
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  >
    <TouchableHighlight
      accessibilityLabel="Dismiss prompt to rate"
      accessibilityTraits="button"
      activeOpacity={0.85}
      onPress={[Function]}
      style={
        Object {
          "padding": 12,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="prompt-to-rate.butotn.dismiss"
      underlayColor="transparent"
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/x-icon.png",
          }
        }
        style={
          Object {
            "tintColor": "#8B969E",
            "width": 16,
          }
        }
      />
    </TouchableHighlight>
    <Heading2
      style={
        Object {
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
       
      Are you enjoying Canvas?
       
    </Heading2>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "margin": 8,
      }
    }
  >
    <Button
      accessibilityTraits="button"
      containerStyle={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#8B969E",
            "borderRadius": 4,
            "borderWidth": 0.5,
            "flex": 1,
            "height": 40,
            "margin": 4,
          },
        ]
      }
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#008EE2",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
      testID="prompt-to-rate.button.no"
    >
      No
    </Button>
    <Button
      accessibilityTraits="button"
      containerStyle={
        Object {
          "backgroundColor": "white",
          "borderColor": "#8B969E",
          "borderRadius": 4,
          "borderWidth": 0.5,
          "flex": 1,
          "height": 40,
          "margin": 4,
        }
      }
      onPress={[Function]}
      style={
        Object {
          "color": "#008EE2",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
      testID="prompt-to-rate.button.yes"
    >
      Yes
    </Button>
  </View>
</Component>
`;

exports[`AppRatingPrompt renders 1`] = `
<Component
  contentStyle={
    Array [
      Object {
        "borderColor": "transparent",
        "borderWidth": 0,
      },
    ]
  }
  hideShadow={true}
  style={
    Array [
      Object {
        "padding": 8,
        "width": 320,
      },
      Object {
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  >
    <TouchableHighlight
      accessibilityLabel="Dismiss prompt to rate"
      accessibilityTraits="button"
      activeOpacity={0.85}
      onPress={[Function]}
      style={
        Object {
          "padding": 12,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="prompt-to-rate.butotn.dismiss"
      underlayColor="transparent"
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/x-icon.png",
          }
        }
        style={
          Object {
            "tintColor": "#8B969E",
            "width": 16,
          }
        }
      />
    </TouchableHighlight>
    <Heading2
      style={
        Object {
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
       
      Are you enjoying Canvas?
       
    </Heading2>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "margin": 8,
      }
    }
  >
    <Button
      accessibilityTraits="button"
      containerStyle={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#8B969E",
            "borderRadius": 4,
            "borderWidth": 0.5,
            "flex": 1,
            "height": 40,
            "margin": 4,
          },
        ]
      }
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#008EE2",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
      testID="prompt-to-rate.button.no"
    >
      No
    </Button>
    <Button
      accessibilityTraits="button"
      containerStyle={
        Object {
          "backgroundColor": "white",
          "borderColor": "#8B969E",
          "borderRadius": 4,
          "borderWidth": 0.5,
          "flex": 1,
          "height": 40,
          "margin": 4,
        }
      }
      onPress={[Function]}
      style={
        Object {
          "color": "#008EE2",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
      testID="prompt-to-rate.button.yes"
    >
      Yes
    </Button>
  </View>
</Component>
`;

exports[`AppRatingPrompt renders hidden 1`] = `
<Component
  contentStyle={
    Array [
      Object {
        "borderColor": "transparent",
        "borderWidth": 0,
      },
    ]
  }
  hideShadow={true}
  style={
    Array [
      Object {
        "padding": 8,
        "width": 320,
      },
      Object {
        "height": 0,
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 8,
      }
    }
  >
    <TouchableHighlight
      accessibilityLabel="Dismiss prompt to rate"
      accessibilityTraits="button"
      activeOpacity={0.85}
      onPress={[Function]}
      style={
        Object {
          "padding": 12,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="prompt-to-rate.butotn.dismiss"
      underlayColor="transparent"
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/x-icon.png",
          }
        }
        style={
          Object {
            "tintColor": "#8B969E",
            "width": 16,
          }
        }
      />
    </TouchableHighlight>
    <Heading2
      style={
        Object {
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
       
      Are you enjoying Canvas?
       
    </Heading2>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "margin": 8,
      }
    }
  >
    <Button
      accessibilityTraits="button"
      containerStyle={
        Array [
          Object {
            "backgroundColor": "white",
            "borderColor": "#8B969E",
            "borderRadius": 4,
            "borderWidth": 0.5,
            "flex": 1,
            "height": 40,
            "margin": 4,
          },
        ]
      }
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#008EE2",
            "fontSize": 16,
            "fontWeight": "600",
          },
        ]
      }
      testID="prompt-to-rate.button.no"
    >
      No
    </Button>
    <Button
      accessibilityTraits="button"
      containerStyle={
        Object {
          "backgroundColor": "white",
          "borderColor": "#8B969E",
          "borderRadius": 4,
          "borderWidth": 0.5,
          "flex": 1,
          "height": 40,
          "margin": 4,
        }
      }
      onPress={[Function]}
      style={
        Object {
          "color": "#008EE2",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
      testID="prompt-to-rate.button.yes"
    >
      Yes
    </Button>
  </View>
</Component>
`;
